kyuubi-main/src/test/scala/org/apache/kyuubi/operation/tpcds/TPCDSOutputSchemaSuite.scala

```diff
@@ -0,0 +1,118 @@
+ *  }}}
+ *
+ * To re-generate golden files for this suite:
+ * {{{
+ *   KYUUBI_UPDATE=1 build/mvn -Dtest=none -DwildcardSuites=org.apache.kyuubi.operation.tpcds.TPCDSOutputSchemaSuite test
+ * }}}
+ */
+// scalastyle:on line.size.limit
+class TPCDSOutputSchemaSuite extends WithKyuubiServer with JDBCTestUtils with TPCDSHelper {
+  override protected val conf: KyuubiConf = KyuubiConf()
+  override protected def jdbcUrl: String = getJdbcUrl
+  override def database: String = this.getClass.getSimpleName
+  override def format: String = "hive OPTIONS(fileFormat='parquet')"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    withJdbcStatement() { statement =>
+      statement.execute(s"CREATE DATABASE IF NOT EXISTS $database")
+      tables.foreach { table =>
+        statement.execute(table.create)
+      }
+    }
+  }
+
+  override def afterAll(): Unit = {
+    withJdbcStatement() { statement =>
+      statement.execute(s"DROP DATABASE IF EXISTS $database CASCADE")
+    }
+    super.afterAll()
+  }
+
+  private val regenerateGoldenFiles = sys.env.get("KYUUBI_UPDATE").contains("1")
+  protected val baseResourcePath: Path = {
+    java.nio.file.Paths.get("src", "test", "resources")
+  }
+
+  private def fileToString(file: Path): String = {
+    new String(Files.readAllBytes(file), Charset.forName("UTF-8"))
+  }
+
+  private def runQuery(query: String, goldenFile: Path): Unit = {
+    withJdbcStatement() { statement =>
+      statement.execute(s"USE $database")
+      val result = statement.executeQuery(query)
+      try {
+        val columnTypes = (1 to result.getMetaData.getColumnCount).map { i =>
+          s"${result.getMetaData.getColumnName(i)}:${result.getMetaData.getColumnTypeName(i)}"
+        }.mkString("struct<", ",", ">\n")
+        if (regenerateGoldenFiles) {
+          Files.write(goldenFile, columnTypes.getBytes())
+        }
+
+        val expected = fileToString(goldenFile)
+        assert(columnTypes === expected)
+      } finally {
+        result.close()
+      }
+    }
+  }
+
+  private def runQueries(name: String): Unit = {
+    val queriesRoot = Thread.currentThread().getContextClassLoader.getResource(name)
+    val queries = Files.list(Paths.get(queriesRoot.toURI))
+    import scala.collection.JavaConverters._
+
+    val validQueries = queries.iterator().asScala.filter { query =>
+      Files.list(query).iterator().asScala.exists { q =>
+        q.getFileName.toString == s"${query.getFileName}.sql"
+      }
+    }
+
+    validQueries.foreach { q =>
+      test(q.getFileName.toString) {
+        val queryFile = Paths.get(q.toString, s"${q.getFileName}.sql")
+        val schemaFile = Paths.get(
+          baseResourcePath.toFile.getAbsolutePath,
+          name,
+          q.getFileName.toString,
+          s"${q.getFileName}.output.schema"
+        )
+        val queryString = fileToString(queryFile)
+        runQuery(queryString, schemaFile)
+      }
+    }
+  }
+
+  runQueries("tpcds-1_4")
```

Comment:
could we refer https://github.com/apache/spark/blob/7242d7f774b821cbcb564c8afeddecaf6664d159/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L220-L244 to make it automatically?